src/blob/core.rs

```diff
@@ -320,14 +316,45 @@ impl Blob {
         self.name.id
     }
 
-    pub(crate) fn check_bloom(&self, key: &[u8]) -> Option<bool> {
+    pub(crate) async fn check_bloom(&self, key: &[u8]) -> Result<Option<bool>> {
         trace!("check bloom filter");
-        self.index.check_bloom_key(key)
+        self.index.check_bloom_key(key).await
+    }
+
+    pub(crate) async fn check_bloom_simple(&self, key: &[u8]) -> Option<bool> {
+        match self.check_bloom(key).await {
```

Comment:
Since you don't check error, you can do the same with: `self.check_bloom(key).await.ok()`